test/test_checksums.py

```diff
@@ -0,0 +1,100 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0.
+
+
+from test import NativeResourceTest
+from awscrt import checksums
+import unittest
+
+
+class TestChecksums(NativeResourceTest):
+
+    def test_crc32_zeros_one_shot(self):
+        output = checksums.crc32(bytes(32))
+        expected = 0x190A55AD
+        self.assertEqual(expected, output)
+
+    def test_crc32_zeros_iterated(self):
+        output = 0
+        for i in range(32):
+            output = checksums.crc32(bytes(1), output)
+        expected = 0x190A55AD
+        self.assertEqual(expected, output)
+
+    def test_crc32_values_one_shot(self):
+        output = checksums.crc32(''.join(chr(i) for i in range(32)))
+        expected = 0x91267E8A
+        self.assertEqual(expected, output)
+
+    def test_crc32_values_iterated(self):
+        output = 0
+        for i in range(32):
+            output = checksums.crc32(chr(i), output)
+        expected = 0x91267E8A
+        self.assertEqual(expected, output)
+
+# stress test gil optimization for 32 bit architecture which cannot handle hug buffer
+    def test_crc32_large_buffer(self):
+        try:
+            giant_buffer = bytes(25 * 2**20)
+        except (MemoryError, OverflowError):
+            raise unittest.SkipTest('Machine cant allocate giant buffer for giant buffer test')
```

Comment:
trivial: remove error handling here. any machine should be able to do a buffer this size